src/overlays/actors/ovl_Bg_Ikana_Dharma/z_bg_ikana_dharma.c

```diff
@@ -28,47 +34,221 @@ const ActorInit Bg_Ikana_Dharma_InitVars = {
+        velocity.z = speed * dirZ;
+        accel.x = velocity.x * -0.05f;
+        accel.y = -0.15f;
+        accel.z = velocity.z * -0.05f;
+        EffectSsKiraKira_SpawnSmallYellow(globalCtx, &pos, &velocity, &accel);
+    }
+}
+
+void BgIkanaDharma_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgIkanaDharma* this = THIS;
+    GlobalContext* globalCtx2 = globalCtx;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    this->dyna.actor.scale.x = 0.3f;
+    this->dyna.actor.scale.y = 0.1f;
+    this->dyna.actor.scale.z = 0.3f;
+    DynaPolyActor_Init(&this->dyna, 0);
+    DynaPolyActor_LoadMesh(globalCtx, &this->dyna, &object_ikana_obj_Colheader_000C50);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->dyna.actor, &sCylinderInit);
+    if (BGIKANADHARMA_IS_CHILD(&this->dyna.actor) == 0) {
+        f32 segmentY = this->dyna.actor.world.pos.y;
+        s32 i = 0;
+        s32 numSegments = BGIKANADHARMA_NUM_SEGMENTS(&this->dyna.actor);
+        for (; i < numSegments; i++) {
+            segmentY += 60.0f;
+            Actor_SpawnAsChildAndCutscene(&globalCtx2->actorCtx, globalCtx, ACTOR_BG_IKANA_DHARMA,
+                                          this->dyna.actor.world.pos.x, segmentY, this->dyna.actor.world.pos.z,
+                                          this->dyna.actor.shape.rot.x, this->dyna.actor.shape.rot.y,
+                                          this->dyna.actor.shape.rot.z, BGIKANADHARMA_FLAG_IS_CHILD,
+                                          this->dyna.actor.cutscene, this->dyna.actor.unk20, NULL);
+        }
+        this->dyna.actor.bgCheckFlags |= 1;
+    }
+    BgIkanaDharma_BeginNormalState(this);
+}
+
+void BgIkanaDharma_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgIkanaDharma* this = THIS;
+
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+    if (sFirstHitBgIkanaDharma == this) {
+        sFirstHitBgIkanaDharma = NULL;
+    }
+}
+
+void BgIkanaDharma_BeginNormalState(BgIkanaDharma* this) {
+    this->actionFunc = BgIkanaDharma_UpdateNormalState;
+    this->dyna.actor.speedXZ = 0.0f;
+}
+
+void BgIkanaDharma_UpdateNormalState(BgIkanaDharma* this, GlobalContext* globalCtx) {
+    s32 wasHit = (this->collider.base.acFlags & AC_HIT) != 0;
+    Player* player = GET_PLAYER(globalCtx);
+    BgIkanaDharma* this2 = this;
+    s32 temp_v0_3;
+
+    if (wasHit) {
+        this->collider.base.acFlags &= ~AC_HIT;
+    }
+    if (wasHit && sFirstHitBgIkanaDharma == NULL) {
+        s32 temp_v0_2;
+        sFirstHitBgIkanaDharma = this2;
+        Flags_SetSwitch(globalCtx, BGIKANADHARMA_GET_SWITCHFLAG(&this->dyna.actor));
+        temp_v0_3 = (s16)(this->dyna.actor.yawTowardsPlayer + 0x8000);
+        temp_v0_2 = ((s16)(player->actor.shape.rot.y - temp_v0_3) >> 1);
+        this->dyna.actor.world.rot.y = temp_v0_3 + temp_v0_2 + 0xF000;
```

Comment:
I wonder if these temps can be replaced with the BINANG macros. Overall I'm a bit skeptical about a lot of the temp variables in this function. Could you make a decomp.me for this function so we can mess with it?